### DISCUSSION TITLE
Gif - Png Same Code

### DISCUSSION THREAD

**Discussion by nirav-kotecha** [SDWebImage/SDWebImage#3212] [Ideas]
Hello,

can we use same code or method or function for download .png, .jpg and .gif images ?

because i am getting url from server and they doesn't give extension of image so i don't know that i have to call function/method for .jpg, .png or .gif

Thanks.

**Comment by dreampiggy** [comment]
Just use `[UIImage sd_imageWithData:]`

**Reply by nirav-kotecha** [reply]
okay. is it take more time to load image if its large size than ?

**Reply by shibusalim** [reply]
Good
